log/src/checks/tcp.ts

```
@@ -0,0 +1,65 @@
+export class TCPChecker {
+    constructor(private readonly siteName: string, private readonly healthEndpoint: string,
+        private readonly logger: ActionLogger) {
+        this.logger.info(`Created Status Checker for ${siteName}`);
+    }
+
+    async verifyEndpoint(): Promise<boolean> {
+        const [host, portString] = this.healthEndpoint.replace('tcp://', '').split(':');
+
+        // Convert the port part to a number
+        const port = parseInt(portString, 10);
+        if (isNaN(port) || port < 1 || port > 65535) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)